File: CSharp/OpenTraceability.GDST/Events/KDEs/VesselCatchInformation.cs
using OpenTraceability.Interfaces;
using OpenTraceability.Utility;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.GDST.Events.KDEs
{
    public class VesselCatchInformation
    {
        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "catchArea")]
        [OpenTraceabilityJson("cbvmda:catchArea")]
        public string? CatchArea { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "economicZone")]
        [OpenTraceabilityJson("cbvmda:economicZone")]
        public string? EconomicZone { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "fishingGearTypeCode")]
        [OpenTraceabilityJson("cbvmda:fishingGearTypeCode")]
        public string? GearType { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "vesselFlagState")]
        [OpenTraceabilityJson("cbvmda:vesselFlagState")]
        public Country? VesselFlagState { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "vesselID")]
        [OpenTraceabilityJson("cbvmda:vesselID")]
        public string? VesselID { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "vesselName")]
        [OpenTraceabilityJson("cbvmda:vesselName")]
        public string? VesselName { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "fisheryImprovementProject")]
        [OpenTraceabilityJson("gdst:fisheryImprovementProject")]
        public string? FIP { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "gpsAvailability")]
        [OpenTraceabilityJson("gdst:gpsAvailability")]
        public bool GPSAvailability { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "imoNumber")]
        [OpenTraceabilityJson("gdst:imoNumber")]
        public string? IMONumber { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "rfmoArea")]
        [OpenTraceabilityJson("gdst:rfmoArea")]
        public string? RFMO { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "satelliteTrackingAuthority")]
        [OpenTraceabilityJson("gdst:satelliteTrackingAuthority")]
        public string? SatelliteTrackingAuthority { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "subnationalPermitArea")]
        [OpenTraceabilityJson("gdst:subnationalPermitArea")]
        public string? SubNationalPermitArea { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "vesselPublicRegistry")]
        [OpenTraceabilityJson("gdst:vesselPublicRegistry")]
        public string? VesselPublicRegistry { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "vesselTripDate")]
        [OpenTraceabilityJson("gdst:vesselTripDate")]
        public DateTimeOffset? VesselTripDate { get; set; }

        [OpenTraceabilityExtensionElements]
        public List<IEventKDE> KDEs { get; set; } = new List<IEventKDE>();
    }
}

File: Java/opentraceability/src/main/java/opentraceability/gdst/events/kdes/VesselCatchInformation.java
package opentraceability.gdst.events.kdes;

import opentraceability.Constants;
import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.Country;
import opentraceability.utility.attributes.*;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

public class VesselCatchInformation {
    @OpenTraceabilityJsonAttribute(name = "cbvmda:catchArea")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "catchArea")
    private String catchArea;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:economicZone")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "economicZone")
    private String economicZone;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:fishingGearTypeCode")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "fishingGearTypeCode")
    private String gearType;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:vesselFlagState")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "vesselFlagState")
    private Country vesselFlagState;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:vesselID")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "vesselID")
    private String vesselID;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:vesselName")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "vesselName")
    private String vesselName;

    @OpenTraceabilityJsonAttribute(name = "gdst:fisheryImprovementProject")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "fisheryImprovementProject")
    private String FIP;

    @OpenTraceabilityJsonAttribute(name = "gdst:gpsAvailability")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "gpsAvailability")
    private boolean GPSAvailability;

    @OpenTraceabilityJsonAttribute(name = "gdst:imoNumber")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "imoNumber")
    private String IMONumber;

    @OpenTraceabilityJsonAttribute(name = "gdst:rfmoArea")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "rfmoArea")
    private String RFMO;

    @OpenTraceabilityJsonAttribute(name = "gdst:satelliteTrackingAuthority")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "satelliteTrackingAuthority")
    private String satelliteTrackingAuthority;

    @OpenTraceabilityJsonAttribute(name = "gdst:subnationalPermitArea")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "subnationalPermitArea")
    private String subNationalPermitArea;

    @OpenTraceabilityJsonAttribute(name = "gdst:vesselPublicRegistry")
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "vesselPublicRegistry")
    private String vesselPublicRegistry;

    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "vesselTripDate")
    private OffsetDateTime vesselTripDate;

    @OpenTraceabilityExtensionElementsAttribute
    private List<IEventKDE> KDEs = new ArrayList<>();

    // Add constructor, getters, and setters here
}
